src/gui/guiFormSpecMenu.cpp

```diff
@@ -3028,29 +3028,31 @@ bool GUIFormSpecMenu::preprocessEvent(const SEvent& event)
 				core::position2d<s32>(event.MouseInput.X, event.MouseInput.Y));
 		if ((hovered) && (hovered->getType() == irr::gui::EGUIET_EDIT_BOX)) {
 			bool retval = hovered->OnEvent(event);
-			if (retval) {
+			if (retval)
 				Environment->setFocus(hovered);
-			}
-			m_JavaDialogFieldName = getNameByID(hovered->getID());
+
+			std::string field_name = getNameByID(hovered->getID());
+			/* read-only field */
+			if (field_name == "")
```

Comment:
please change this by field_name.empty() it's more performant

Reply:
Change below also?

Reply:
yeah too that can be nice

Reply:
Done.